### PR TITLE
Contract page token transfers - Issue #40

### PR DESCRIPTION
**Pull request by melnikga**
#40 
Added the ERC-20 token transfers table

### CODE REVIEW

File: package.json
Comment (saimeunt) [inline package.json:1]: Please undo changes to this file, `cross-env` should no longer be needed.

File: src/components/pages/address/address-token-transfers.tsx [hunk 1]
```diff
@@ -1,11 +1,123 @@
...
+                      <TooltipTrigger>
+                        {item.transactionHash.slice(0,12)}...
+                      </TooltipTrigger>
+                      <TooltipContent>
+                        {item.transactionHash}
+                      </TooltipContent>
+                    </Tooltip>
+                  </TableCell>
+                <TableCell>
+                  {item.method}
```
Comment (saimeunt) [inline src/components/pages/address/address-token-transfers.tsx]: We have a component for displaying methods, please use `TxMethodBadge`.

File: src/lib/fetch-data.ts [hunk 2]
```diff
@@ -209,3 +212,63 @@ export const fetchTokensPrices = async () => {
...
+    });
+
+    const allTransfers = await Promise.all(erc20Transfers.map(async t => {
+      const transactionDate = new Date(Number(t.receipt.transaction.block.timestamp) * 1000);
+
+      return {
+        transactionHash: t.transactionHash,
+        method: t.receipt.transaction.input.slice(0, 10),
+        block: Number(t.receipt.transaction.blockNumber),
+        age: formatDistanceToNow(transactionDate, { addSuffix: true }),
```
Comment (saimeunt) [inline src/lib/fetch-data.ts]: You can keep the raw timestamp here and use `formatTimestamp` in the table instead.

File: src/lib/fetch-data.ts [hunk 3]
```diff
@@ -209,3 +212,63 @@ export const fetchTokensPrices = async () => {
...
+  page: number = 1,
+  limit: number = 100
+) {
+  console.log(contractAddress);
+  const skip = (page - 1) * limit;
+
+  try {
+    const erc20Transfers = await prisma.erc20Transfer.findMany({
+      where: {
+        address: contractAddress,
```
Comment (saimeunt) [inline src/lib/fetch-data.ts]: Should be `OR: [{ from: contractAddress }, { to: contractAddress }]` instead.

File: src/lib/fetch-data.ts [hunk 4]
```diff
@@ -209,3 +212,63 @@ export const fetchTokensPrices = async () => {
...
+      const transactionDate = new Date(Number(t.receipt.transaction.block.timestamp) * 1000);
+
+      return {
+        transactionHash: t.transactionHash,
+        method: t.receipt.transaction.input.slice(0, 10),
+        block: Number(t.receipt.transaction.blockNumber),
+        age: formatDistanceToNow(transactionDate, { addSuffix: true }),
+        from: t.from,
+        to: t.to,
+        amount: formatEther(BigInt(t.value)),
```
Comment (saimeunt) [inline src/lib/fetch-data.ts]: This should be formatted with `formatUnits`.

**Review by saimeunt — CHANGES_REQUESTED** [review]
@melnikga this is mostly correct, please make the requested changes and make sure there's no conflict with main before we can merge.

**Comment by melnikga** [comment]
@saimeunt Necessary changes have been made

**Review by saimeunt — APPROVED** [review]

**Pull request merged**